### ISSUE TITLE
 Pinetime: wasp-reloader - use new Infinitime bootloader 1.0.1 for building factory image

### ISSUE THREAD

**Issue by GaryM0101** [wasp-os/wasp-os#519]
A new [bootloader for Infinitime](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader/releases/tag/1.0.1) was patched for the new SPI flash memory chip that was put in for the new batch of Pinetime devices that started in September.

We saw a recent issue where a user tried to flash the old bootloader 1.0.0 onto a new Pinetime with the new SPI flash memory chip. They soft-bricked the device.  

The wasp-reloader should use the new Infinitime bootloader 1.0.1 for building factory image (reloader-factory.zip)

Folks wanting to go back to Infinitime with the new Pinetime SPI chip will undoubtedly soft-brick their device in doing so.

**Comment by daniel-thompson** [comment]
I have to admit I have almost no recollection of how any of this stuff works but I went through the wasp-reloader build system until I figured things out.

Instructions on how to update the Infinitime bootloader are here:
https://github.com/wasp-os/wasp-reloader/blob/14f85fc92180c0b29f36400480b54a571dc510b8/src/boards/pinetime/factory_image.h

**Comment by GaryM0101** [comment]
I built a test version of the factory-reloader.zip with the new 1.0.1 bootloader and successfully deployed it on an older devkit. The reloader still uses the pinetime-mcuboot-recovery-image-0.11.0. We'll still need to test it for the new SPI chip and formally get a PR going to deploy it.

Not sure if any other changes will be needed for WaspOS but we'll find out here shortly. For now I'll attach the updated file here for folks that might need it in case they didn't see the warnings.

[reloader-factory.zip](https://github.com/user-attachments/files/18390367/reloader-factory.zip)

**Comment by GaryM0101** [comment]
The new Wasp "reloader" tool that flashes the InfiniTime bootloader v 1.0.1 was successfully tested on the pinetime with the new BY25Q32 memory chip the TL Lim shipped to me last week. 

We'll get a PR going to get the new factory_image.h into the release but folks can now use the [reloader-factory.zip](https://github.com/user-attachments/files/18390367/reloader-factory.zip) to revert back to the pinetime-mcuboot-recovery-image and flash the latest Infinitime firmware.

From the Wasp side I had no issues flashing the reloader-mcuboot and the micropython v0.4 firmware. I was able to access the SPI flash area without any issues.

**Comment by pawel-ch** [comment]
So should we still refrain from switching OSes as per https://wiki.pine64.org/wiki/Switching_your_PineTime_between_InfiniTime_and_Wasp-os and https://pine64.org/2024/12/29/please-refrain-from-installing-waspos-on-new-pinetimes-for-the-moment/? Or maybe not updating the documentation and not publishing a new blog post about this was an oversight?

**Comment by chewi** [comment]
I wonder what happened to me today then. I saw the warning about 1.0.1, saw HW 1.0.0 and FW 1.15 on my device, and figured that meant I wasn't affected. It's my kid's watch, so I hadn't seen it reboot recently. It wasn't until after I flashed reloader-mcuboot.zip that I saw the green pinecone with 1.0.1 and my heart sank. I got this device a year ago, and I now understand that this is the current bootloader version. So the blue pinecone with the "Start" arrow came up, but Gadgetbridge wouldn't see it. I thought it was game over. Eventually, I tried adding it as an unsupported device, flashed micropython.zip, and I was back in business. So I don't know if it really was bricked or not. This was the 0.4.1 release from 2021, so it's not like it had a fix. Was I just lucky?

**Comment by GaryM0101** [comment]
You got lucky. As long as you use the reloader-factory.zip in the above message to revert back to Infinitime, there should be no problem.

**Comment by positron96** [comment]
Hi everyone. I am coming here from https://wiki.pine64.org/wiki/Switching_your_PineTime_between_InfiniTime_and_Wasp-os, where a big red banner now says to not flash wasp-os over infinitime with bootloader 1.0.1, which is what my PineTime comes with. 
But, after reading through this thread, I understand that it is resolved, right?

Could you give more clarifications on the process? The `reloader-factory` file above shall be used when moving from wasp-os back to infinitime, right? To move from infinitime to wasp-os I need to use `reloader-mcuboot` from official documentation?

**Comment by chewi** [comment]
I was told that it isn't resolved. I was somehow just lucky. I wouldn't risk it if I were you.

**Comment by GaryM0101** [comment]
Yes the reloader-factory is to be [used above](https://github.com/wasp-os/wasp-os/issues/519#issuecomment-2585841213) when going back.

**Comment by positron96** [comment]
Well, if that's what fixes the problem, I made 0.4.1 release archive with reloader-factory.zip from above. It can be put into releases page (or replace the existing archive). 

[wasp-os-0.4.1-with-factory1.0.1.zip](https://github.com/user-attachments/files/26962720/wasp-os-0.4.1-with-factory1.0.1.zip)

**Comment by schenectadyboy12308-jpg** [comment]
Ohhhh is this why mine is softbricked because I used the factory zip
